### PR TITLE
Wilma Responsive redesign

### PR DESCRIPTION
**Pull request by milospp**
**[GitHub issue 3910](https://github.com/vivo-project/VIVO/issues/3910)**
**[GitHub PR Vitro 460](https://github.com/vivo-project/Vitro/pull/460)**

# What does this pull request do?
Added bootstrap (only grid style) CSS and modified Wilma theme to leverage its responsive design feature and enhance the overall layout.

# What's new?
Mobile optimized design
Avoided horizontal scroll

# How should this be tested?
Open every page and test by resizing up to around 400px in width (Mobile device toolbar can be used in google chrome to simulate phone display)

# Additional Notes:
Here is the video example

- The Left window is VIVO with an old Wilma theme
- The middle window is a new, updated Wilma theme
- The right window is Tenderfoot theme (just for comparison)

[Video Example](https://youtu.be/9B0_cevrEeE)
https://youtu.be/9B0_cevrEeE

### CODE REVIEW

**Comment by gneissone** [comment]
@milospp took this for a test spin and works great overall. Note for future reviewers, it is necessary to do a hard refresh to clear your cache of the old wilma.css! I was fooled for a few minutes.

**Review by balmas — COMMENTED** [review]
The responsiveness looks good to me. Tested in Chrome, FF and Safari.

However, although I cannot see any reason for this in the code, so it could be something caused by environment, I cannot load any of the temporal graph google charts when I build from the PR code. I get a cross-site scripting failure from the googlecharts API and jquery query error for the chart height. (see attached for example)
English-Temporal-Graph-Visualization (1)

I cannot understand why this is only the case in the newer code, but it is consistently reproducible for me. Apologies if this is something I'm doing wrong.

**Review by balmas — APPROVED** [review]
Given that the chart issue is already fixed in main, and should be handled by the merge, I think this change looks good.

File: webapp/src/main/webapp/themes/nemo/templates/head.ftl
Comment (litvinovg) [inline webapp/src/main/webapp/themes/nemo/templates/head.ftl:23]: It seems bootstrap5 and bootstrap3 would be loaded at the same time according to head.ftl and headScripts.ftl of Nemo theme.

File: webapp/src/main/webapp/js/individual/propertyGroupControls.js
Comment (milospp) [reply webapp/src/main/webapp/js/individual/propertyGroupControls.js]: Removed, thanks for the suggestion.

File: webapp/src/main/webapp/themes/wilma/templates/page-home.ftl
Comment (milospp) [reply webapp/src/main/webapp/themes/wilma/templates/page-home.ftl]: It's been a while since I added this code, I can't remember why I switched to the POST method... Also, I see that every other theme is using GET.
I reverted it to GET.

**Comment by gneissone** [comment]
Tried out the changes and looks good. @milospp, I did notice some odd behavior if the geo focus map was enabled in runtime.properties (homePage.geoFocusMaps=enabled). See screenshot:
Screenshot 2024-07-15 at 14 49 50

**Comment by milospp** [comment]
> Tried out the changes and looks good. @milospp, I did notice some odd behavior if the geo focus map was enabled in runtime.properties (homePage.geoFocusMaps=enabled). See screenshot: Screenshot 2024-07-15 at 14 49 50

@gneissone, I fixed the map responsiveness. Can you please validate it.

image

**Review by gneissone — APPROVED** [review]
Looks good after the update, thank you!

**Comment by balmas** [comment]
Just confirming that I agree this still looks good after the change to use the bootstrap webjar.

**Review by litvinovg — CHANGES_REQUESTED** [review]
This set of PRs not only modifies Wilma theme to have responsive design, but also introduces global dependency for shared components on specific version of Bootstrap which can't safely be done retrospectively.
So it breaks backward compatibility for customized themes
https://wiki.lyrasis.org/display/VIVODOC115x/Creating+a+custom+theme
I think we should find a better way to do that.

**Comment by chenejac** [comment]
@milospp we concluded at a committers meeting the following:

- This should be a new theme called willow
- Changes of shared files in this PR (e.g. webapp/src/main/webapp/css/*) should be moved under the theme directory if it is possible

Can you please align this PR in accordance with those conclusions? Thanks

**Comment by gneissone** [comment]
@chenejac is it also reasonable to change the [default theme](https://github.com/vivo-project/VIVO/blob/main/installer/example-settings.xml#L14) to 'willow' as part of the PR?

**Comment by chenejac** [comment]
> @chenejac is it also reasonable to change the [default theme](https://github.com/vivo-project/VIVO/blob/main/installer/example-settings.xml#L14) to 'willow' as part of the PR?

General idea is that willow will be the default VIVO theme. The team will make the final decision when this PR is close to be merged.

**Comment by hauschke** [comment]
Just a quick note regarding the name: AFAIK, Wilma and Tenderfoot were named after... ["Wilma Tenderfoot"](https://www.penguinrandomhouse.com/series/DW0/wilma-tenderfoot/). In that tradition it could make sense to call it "Pickle", "Bludsten", or "Theodore".  But that just as a sidenote, I have no strong feelings here.

**Comment by brianjlowe** [comment]
Wilma was named after the first granddaughter of Jon Corson-Rikert, the IT director at Mann Library and original creator of VIVO.

**Comment by hauschke** [comment]
> Wilma was named after the first granddaughter of Jon Corson-Rikert, the IT director at Mann Library and original creator of VIVO.

Ah, that's cute and new to me. Okay, so (Wilma) Tenderfoot has nothing to to with the original Wilma. Thanks for the correction! But I am sure that Tenderfoot was named after Wilma Tenderfoot.

**Comment by brianjlowe** [comment]
I had no idea where Tenderfoot came from, but that certainly makes sense if someone reinterpreted Wilma in a new context.

**Comment by gneissone** [comment]
re:https://github.com/vivo-project/VIVO/pull/3972#pullrequestreview-2071297157

I am still seeing the issue with the temporal graphs after merging in the main branch, unfortunately. The sparkline graph works fine. @milospp any ideas what could be going wrong? I tried manually setting a width for the div holding the graph but there was no change.

**Review by gneissone — CHANGES_REQUESTED** [review]
Temporal graph issue needs resolved.

File: webapp/src/main/webapp/themes/wilma/css/wilma.css
Comment (gneissone) [inline webapp/src/main/webapp/themes/wilma/css/wilma.css:2290]: I suspect this is the problem. If this #graphContainer css is removed then the temporal graph works again.

**Pull request closed without merging**
